ghcide/test/exe/Main.hs

```diff
@@ -5553,12 +5553,17 @@ bootTests = testGroup "boot"
         -- Dirty the cache
         liftIO $ runInDir dir $ do
             cDoc <- createDoc cPath "haskell" cSource
-            _ <- getHover cDoc $ Position 4 3
-            ~() <- skipManyTill anyMessage $ satisfyMaybe $ \case
-                FromServerMess (SCustomMethod "ghcide/reference/ready") (NotMess NotificationMessage{_params = fp}) -> do
-                    A.Success fp' <- pure $ fromJSON fp
-                    if equalFilePath fp' cPath then pure () else Nothing
-                _ -> Nothing
+            let hoverParams = HoverParams cDoc (Position 4 3) Nothing
+            hoverRequestId <- sendRequest STextDocumentHover hoverParams
+            inBetweenMsgs <- manyTill anyMessage (responseForId STextDocumentHover hoverRequestId)
+            let isReadyMsg = \case
+                  FromServerMess (SCustomMethod "ghcide/reference/ready") (NotMess NotificationMessage{_params = params}) ->
+                    case fromJSON params of
+                      A.Success fp | equalFilePath fp cPath -> Just ()
+                      _ -> Nothing
+                  _ -> Nothing
```

Comment:
The idiomatic thing here is to skip all messages except for `Either` the hover response or the reference ready notification.